hack/jenkins/e2e-runner.sh

```diff
@@ -186,6 +179,37 @@ function dump_cluster_logs() {
     fi
 }
 
+# GCI specific settings.
+# Assumes: JENKINS_GCI_IMAGE_FAMILY
+function setup_gci_vars() {
+  local -r gci_staging_project=container-vm-image-staging
+  local -r image_info="$(gcloud compute images describe-from-family ${JENKINS_GCI_IMAGE_FAMILY} --project=${gci_staging_project} --format=json)"
+  local -r image_description="$(echo ${image_info} | jq '.description')"
+  local -r image_name="$(echo ${image_info} | jq '.name')"
+
+  if [[ "${JENKINS_USE_GCI_VERSION:-}" =~ ^[yY]$ ]]; then
+    # GCI QA jobs use the builtin k8s version.
+    # Staged GCI images all include versions in their image descriptions so we
+    # extract builtin Kubernetes version from them.
+    local -r k8s_version_re='.*Kubernetes: ([0-9a-z.-]+),.*'
+    if [[ ${image_description} =~ ${k8s_version_re} ]]; then
+      export JENKINS_PUBLISHED_VERSION="release/${BASH_REMATCH[1]}"
```

Comment:
If you have an alternative in the works, I'm fine with the current version as an interim solution.